messaging/src/main/java/org/axonframework/messaging/unitofwork/SimpleProcessingContext.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2010-2024. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging.unitofwork;
+
+import jakarta.annotation.Nonnull;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.function.UnaryOperator;
+
+/**
+ * Implementation of the {@link ProcessingContext} that can only store resources.
+ * Registering a lifecycle action will not take effect.
+ *
+ * @author Allard Buijze
+ * @since 5.0.0
+ */
+public class SimpleProcessingContext implements ProcessingContext {
```

Comment:
I always prefer a name that means something, sometimes it's diff ult to even find one proposition, but for this I propose something like:
 - `OnlyResourcesProcessingContext` 
 - `JustResourcesProcessingContext`
 - `ResourcesPassingProcessingContext`. 
 
 Simple in other parts of the framework most of the time it's something useful, like basic/default implementation of the interface, in this case it lacks some parts of the implementation.

Reply:
Hmm... good question. I'm not sure. Looking at just `empty()` I expect the ProcessingContext which may be fully-fledged - I mean just empty, but I expect all methods like configuring lifecycle handlers to work. 
So definitiely some adjustment here is needed.